etcdserver/api/v2http/peer.go

```diff
@@ -31,8 +31,10 @@ const (
 // NewPeerHandler generates an http.Handler to handle etcd peer requests.
 func NewPeerHandler(s *etcdserver.EtcdServer) http.Handler {
 	var lh http.Handler
-	if l := s.Lessor(); l != nil {
-		lh = leasehttp.NewHandler(l)
+	l := s.Lessor()
+	w := s.ApplyWait()
+	if l != nil && w != nil {
```

Comment:
we will have a nil pointer exception in leaseHTTP. Probably need to add an extra check.